packages/resilient/styles/init.lua

```diff
@@ -88,7 +88,8 @@ end
 
 function package:readStyles ()
   local yaml = require("resilient-tinyyaml")
-  local fname = SILE.masterFilename .. '-styles.yml'
+  local basename = pl.path.splitext(pl.path.normpath(SILE.input.filenames[1]))
```

Comment:
So somehow the magic behind ` SILE.masterFilename` got broken and 3rd party packages have to use an even weirder magic `pl.path.splitext(pl.path.normpath(SILE.input.filenames[1]))`?

I suggest reporting the issue to SILE.

Reply:
No, it didn't "get broken." It was *always* broken because it made assumptions about inputs that limited usage. You ran into one such issue yourself in [SILE#1833](https://github.com/sile-typesetter/sile/issues/1833). In working on a fix I ran into several other issues and also raised the point that `masterFilename` is not even a file name at all (c.f. [SILE#1835](https://github.com/sile-typesetter/sile/pull/1835)). It also has never properly handled STDIN or named pipes (something we added in SILE v0.9.5) because it can't invent a name, and more recently it became obvious that it doesn't play nice with multiple inputs.

We disagreed on what the most useful abstraction was and what to call it, so rather than remove the poorly named and partially functional API we've just left it for now but started using more robust code internally. There is no reason for 3rd party packages to not adopt this until such a time a a standardized base name function of some kind is provided that covers all these bases.

Normalizing a path and getting the base name is not exactly weird magic, but this code is making assumptions about projects using certain file system layouts (and even using real files instead of named pipes, anonymous streams, command substitution, etc.). It's okay to make those assumptions for this use case, but SILE can't magically do the right thing in every case so explicitly doing exactly the thing you want to accomplish here makes sense to me.